### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by hexaglow**
Unit tests are now enabled, using Catch2 for testing and Trompeloeil for mocking.
Tests are run automatically by the CI system.

At the moment it's just a couple of simple tests on `AST::Node`, this provides a template for how to add more tests.

I've made a couple of other changes in other areas, to ensure the CI is happy with everything.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,11 +38,30 @@ if ($ENV{CLION_IDE})
 endif ()
 target_compile_definitions(llvm_interface INTERFACE ${LLVM_DEFINITIONS})
 
-if ($ENV{CLION_IDE})
-    message(STATUS "Using extra llvm include files as CLion can't follow WSL symlinks")
-    target_include_directories(llvm_interface INTERFACE "/usr/include/llvm-10")
-endif ()
```
Comment (the-Bruce) [inline CMakeLists.txt:43]: I think that these need to continue to be present for the people using windows

File: src/parser/src/Type.cpp [hunk 2]
```diff
@@ -59,3 +59,7 @@ llvm::Type* Typing::MatrixType::getLLVMType(Utils::IRContext* context) const {
     }
     return ty;
 }
+const std::vector<uint>& Typing::MatrixType::getDimensions() const { return this->dimensions; }
+
+// TODO make this not a noop
+bool Typing::MatrixType::simpleDimensionCompatible(const Typing::MatrixType& val) const { return true; }
```
Comment (the-Bruce) [inline src/parser/src/Type.cpp:65]: I was keeping these in the header to make it more obvious that they aren't functional yet...

File: src/parser/src/Type.cpp [hunk 3]
```diff
@@ -29,7 +29,7 @@ int Typing::MatrixType::getLength() const {
     return std::accumulate(this->dimensions.begin(), this->dimensions.end(), 1, std::multiplies());
 }
 llvm::Type* Typing::MatrixType::getLLVMType(Utils::IRContext* context) const {
-    llvm::Type* ty;
+    llvm::Type* ty = nullptr;
```
Comment (the-Bruce) [inline src/parser/src/Type.cpp:32]: I personally prefer the uniform initialiser syntax `llvm::Type* ty {};`, but comes to the same effect

**Review by the-Bruce — APPROVED** [review]
Few stylistic comments (although nothing blocking merge), but otherwise all looks solid

File: CMakeLists.txt (hunk 1 above)
Comment (hexaglow) [reply CMakeLists.txt:43]: That is still in the file actually, it just got moved somehow

**Pull request merged**